test/rails/config.yml

```diff
@@ -71,23 +71,24 @@ connections:
       prepared_statements: <%= ENV['PREPARED_STATEMENTS'] || 'false' %>
 
   postgresql:
-    arunit:
-      insert_returning: <%= ENV['INSERT_RETURNING'] %>
-      prepared_statements: <%= ENV['PREPARED_STATEMENTS'] %>
+    _setup: &_setup
+      <%- if ENV.key?('INSERT_RETURNING') %>
+      insert_returning: <%= ENV['INSERT_RETURNING'] %> # really should leave it off to no key if not set
+      <%- end %>
+      <%- if ENV.key?('PREPARED_STATEMENTS') %>
+      insert_returning: <%= ENV['PREPARED_STATEMENTS'] %> # should leave it off to its default if not set
+      <%- end %>
       username: <%= ENV['PGUSER'] || ENV['user'] || 'rails' %>
       properties:
-        prepareThreshold: <%= ENV['PREPARE_THRESHOLD'] || 1 %>
+        prepareThreshold: <%= ENV['PREPARE_THRESHOLD'] %>
```

Comment:
We need the prepare threshold to be set to 1 for the tests that check some things with prepared statements otherwise the statements don't get prepared on the server like AR expects them to

Reply:
@kares I do not have pgsql set up but I had to remove 'if' statements from a previous commit because I could not get rails to load the yaml: fb9e3a03d3fc83ed61914a95f49ed4a7d3e4bcc2.  I am not saying this is broken but I don't know why I had a problem then (when I am assuming you did not).

Reply:
@enebo ah I recalled it was in at some point - that is just a bit crazy ... we need to test w defaults

@rdubya so you think that a lot of failures are due this PS = 1? 
felt like the defaults are not handled right ... almost as if postgresql only passes 'reasonably' with the proper configuration in place (not a default configuration most user will have). 
its slightly scary to need to hack things to a 'non' real world setup. but o.k. I can leave it as is ...

Reply:
@kares yeah its a little odd but AR prepares the statement on the server the first time it is used and the postgres jdbc driver defaults to only preparing it on the server after the 5th time the prepared statement object is executed. In real world usage it wouldn't make much of a difference (other than not preparing the statements on the server for statements that are only used as 1 offs)

Reply:
aaah okey - thought its a limit for the statements prepared (upper bound for the driver pool).
... although the 5 vs 1 might still going to end up being interesting in real-world scenario :)

so I will put this back than, thanks

Reply:
@kares @rdubya  hey let's add a comment there explaining it is JDBC not preparing until 5 so we don't have this conversation again in two years :)